src/BenchmarkDotNet/Portability/RuntimeInformation.cs

```diff
@@ -25,7 +25,7 @@ internal static class RuntimeInformation
         internal const string ReleaseConfigurationName = "RELEASE";
         internal const string Unknown = "?";
 
-        public static bool IsMono { get; } = Type.GetType("Mono.Runtime") != null; // it allocates a lot of memory, we need to check it once in order to keep Engine non-allocating!
+        public static bool IsMono { get; } = Type.GetType("Mono.RuntimeStructs") != null; // it allocates a lot of memory, we need to check it once in order to keep Engine non-allocating!
```

Comment:
Does this still return true on the old Mono runtime?

Mono checks are performed elsewhere in the library like in the memory diagnoser (`GcStats`) since it did not support `AppDomain.MonitoringIsEnabled`. Is that still the case with the new Mono runtime?

It's also checked in the `MonoDisassembler`, and I have no idea if that's relevant here.

And other places in RuntimeInformation.cs, all of which should be verified.

Reply:
Ok, I just tested (using Adam's example), and it seems monitoring is still not available on the new Core/Mono runtime. I would assume everything else behaves the same, too (`MonoDisassembler`), but I didn't test.